coveo-stew/coveo_stew/stew.py

```diff
@@ -291,13 +323,38 @@ def poetry_run(
         if breakout_of_venv:
             environment_variables.pop("VIRTUAL_ENV", None)
 
-        return check_run(
-            *poetry_env.build_command(PythonTool.Poetry, *commands, "-vv" if self.verbose else ""),
-            working_directory=self.project_path,
-            capture_output=capture_output,
-            verbose=self.verbose,
-            env=environment_variables,
-        )
+        with self._activate_poetry_environment(environment):
+            return check_run(
+                *poetry_env.build_command(
+                    PythonTool.Poetry, *commands, "-vv" if self.verbose else ""
+                ),
+                working_directory=self.project_path,
+                capture_output=capture_output,
+                verbose=self.verbose,
+                env=environment_variables,
+            )
+
+    @contextmanager
+    def _activate_poetry_environment(
+        self, environment: PythonEnvironment = None
```

Comment:
Should it be `Optional[PythonEnvironment]`? Actually I'm surprise that `mypy` doesn't complain... :thinking: Did you do some magic so that `PythonEnvironment` can be `None`?

Reply:
That's exactly that. Optional is implied when default is None. Compare with the annotion of `_cached_activated_environment`.

Reply:
well actually... seems like this is actually deprecated and we should start putting Optional everywhere... 😱 

https://github.com/python/mypy/issues/9091

Reply:
https://stackoverflow.com/questions/62732402/can-i-omit-optional-if-i-set-default-to-none

Reply:
:facepalm:

Reply:
nonetheless, we'll do them all at the same time. no need to do this here.